discord_bots/utils.py

```diff
@@ -509,9 +509,9 @@ async def send_message(
         print("[send_message] exception:", e)
 
 
-async def print_leaderboard(channel=None):
+async def print_leaderboard(interaction: Optional[Interaction] = None):
     output = "**Leaderboard**"
-    session = Session()
+    session: SQLAlchemySession = Session()
```

Comment:
since you're already fixing up this function, you might as well update the session handling to use the context manager (that doesn't begin a transaction).
```python
session: SQLalchemySession
with Session() as session:
```
It's safe to replace this with it and a bit more robust. You will no longer need to manually close the session, but you do have to remember to `session.commit()` if necessary

Reply:
Thanks for the tip, updated in commit [64f989c](https://github.com/tribesonecommunity/discord-bots/pull/65/commits/64f989cfdc0157357485970dab481e0b31f5b0bb)